orderbook/src/price_estimate.rs

```diff
@@ -0,0 +1,88 @@
+use anyhow::{anyhow, Result};
+use ethcontract::H160;
+use model::TokenPair;
+use shared::uniswap_pool::PoolFetching;
+use std::iter::once;
+
+#[allow(dead_code)]
+struct UniswapPriceEstimator {
+    pool_fetcher: Box<dyn PoolFetching>,
+}
+
+impl UniswapPriceEstimator {
+    // Estimates the price using the direct pool between sell and buy token. Price is given in
+    // how much of sell_token needs to be sold for one buy_token.
+    // Returns an error if no pool exists between sell and buy token.
+    #[allow(dead_code)]
+    pub async fn estimate_price(&self, sell_token: H160, buy_token: H160) -> Result<f64> {
+        let pair = match TokenPair::new(sell_token, buy_token) {
+            Some(pair) => pair,
+            None => return Ok(1.0),
+        };
+        let pool = self
+            .pool_fetcher
+            .fetch(once(pair).collect())
+            .await
+            .pop()
+            .ok_or_else(|| anyhow!("Uniswap pool does not exist"))?;
+        let (sell_reserve, buy_reserve) = if pool.tokens.get().0 == sell_token {
+            (pool.reserves.0, pool.reserves.1)
+        } else {
+            (pool.reserves.1, pool.reserves.0)
+        };
+        Ok((sell_reserve as f64) / (buy_reserve as f64))
```

Comment:
We might want to have a check for the case where one of the reserves is 0. Not strictly necessary I am just wondering if it could lead to problems with how this will be used in the future. If one of the reserves is 0 then the price could be 0 or +inf.

Reply:
Good point! Will filter out those pools